### PR TITLE
chore:[CO-1019] Bulk apply common code clean-up

### PR DESCRIPTION
**Pull request by keshavbhatt**
**Motivation:**

This pull request addresses code cleanliness with focus on enhancing maintainability, readability, and adherence to coding standards. By performing bulk code cleanup, we aim to streamline the codebase, making it more comprehensible for current and future developers. The primary goal is to enhance the overall quality of the project by eliminating redundant, unnecessary, or outdated code applying suggestions provided by IDE in bulk. 

**Notes for reviewers:**
- Each commit strictly represents a single improvement/changeset, please review each commit instead of consolidated change. 
- This change was not splitted into multiple PRs to avoid unnecessary noise and possible merge conflicts.
- commits starting from 5d0a457009d4c4386b9c76b22b7d921761de0d53 are changes we made in peer review session.

### CODE REVIEW

File: client/src/main/java/com/zimbra/client/ZContactHit.java [hunk 1]
```diff
@@ -299,15 +299,17 @@ public String getFullName() {
     public String getFileAs() { return mFileAs; }
     public String getNickname() { return mNickname; }
     public String getNamePrefix() { return mNamePrefix; }
-    public String getFirstName() { return mFirstName; };
-    public String getPhoneticFirstName() { return mPhoneticFirstName; }
+    public String getFirstName() { return mFirstName; }
+
+  public String getPhoneticFirstName() { return mPhoneticFirstName; }
```
Comment (sonersivri) [inline client/src/main/java/com/zimbra/client/ZContactHit.java:304]: @keshavbhatt is corrupted no need yo change this class may be rallback

File: client/src/main/java/com/zimbra/client/ZFolder.java [hunk 2]
```diff
@@ -587,11 +588,7 @@ public String getName() {
   }
 
   public String getNameURLEncoded() {
-    try {
-      return URLEncoder.encode(mName, "utf-8").replace("+", "%20");
-    } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline client/src/main/java/com/zimbra/client/ZFolder.java:593]: @keshavbhatt why delete this check?

File: client/src/main/java/com/zimbra/client/ZFolder.java [hunk 3]
```diff
@@ -638,11 +635,7 @@ public String getPathURLEncoded() {
     else {
       String pp = mParent.getPathURLEncoded();
       String n = null;
-      try {
-        n = URLEncoder.encode(mName, "utf-8").replace("+", "%20");
-      } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline client/src/main/java/com/zimbra/client/ZFolder.java:644]: @keshavbhatt why this is removed

File: common/src/main/java/com/zimbra/common/calendar/ZCalendar.java [hunk 4]
```diff
@@ -1158,11 +1155,7 @@ public static class ZCalendarBuilder {
 
     public static ZVCalendar build(String icalStr) throws ServiceException {
       ByteArrayInputStream bais = null;
-      try {
-        bais = new ByteArrayInputStream(icalStr.getBytes(MimeConstants.P_CHARSET_UTF8));
-      } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/calendar/ZCalendar.java:1163]: @keshavbhatt why remove this check?

File: common/src/main/java/com/zimbra/common/iochannel/Packet.java [hunk 5]
```diff
@@ -57,12 +58,8 @@ public boolean hasRemaining() {
 
     private Packet(String clientId, PeerServer server, ByteBuffer message) {
         byte[] clientIdBytes;
-        try {
-            clientIdBytes = clientId.getBytes("UTF-8");
-        } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/iochannel/Packet.java:62]: @keshavbhatt why this check removed

File: common/src/main/java/com/zimbra/common/iochannel/Packet.java [hunk 6]
```diff
@@ -108,10 +105,8 @@ public static Packet fromBuffer(ByteBuffer buffer) throws IOChannelException {
         buffer.flip().position(copy.position());
         buffer.compact();
         if (log.isDebugEnabled()) {
-            try {
-                log.debug("msg from %s '%s'", new String(headerBuf, "UTF-8"), new String(contentBuf, "UTF-8"));
-            } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/iochannel/Packet.java:113]: @keshavbhatt Why remove this check?

File: common/src/main/java/com/zimbra/common/mime/ContentTransferEncoding.java [hunk 7]
```diff
@@ -39,7 +39,8 @@ ContentTransferEncoding normalize() {
     static class Base64EncoderStream extends TransferEncodingStream {
         static final byte[] BASE64_TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".getBytes();
 
-        private int column, buf[] = new int[4];
+        private int column;
+      private int[] buf = new int[4];
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/mime/ContentTransferEncoding.java:43]: @keshavbhatt format problem

File: common/src/main/java/com/zimbra/common/mime/ContentTransferEncoding.java [hunk 8]
```diff
@@ -84,7 +85,9 @@ static class Base64DecoderStream extends TransferEncodingStream {
             }
 
         private boolean closed;
-        private int position = 3, valid = 3, buf[] = new int[3];
+        private int position = 3;
+      private int valid = 3;
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/mime/ContentTransferEncoding.java:89]: @keshavbhatt format problem

File: common/src/main/java/com/zimbra/common/util/ByteUtil.java [hunk 9]
```diff
@@ -472,20 +473,12 @@ public static String encodeFSSafeBase64(byte[] data) {
             if (encoded[i] == (byte) '/')
                 encoded[i] = (byte) ',';
         }
-        try {
-            return new String(encoded, "utf-8");
-        } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/ByteUtil.java:482]: @keshavbhatt check the remove

File: common/src/main/java/com/zimbra/common/util/ByteUtil.java [hunk 10]
```diff
@@ -472,20 +473,12 @@ public static String encodeFSSafeBase64(byte[] data) {
...
-            return null;  // this shouldn't happen
-        }
+        return new String(encoded, StandardCharsets.UTF_8);
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/ByteUtil.java:476]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/ByteUtil.java [hunk 11]
```diff
@@ -472,20 +473,12 @@ public static String encodeFSSafeBase64(byte[] data) {
...
     }
 
     public static byte[] decodeFSSafeBase64(String str) {
         byte[] bytes = null;
-        try {
-            bytes = str.getBytes("utf-8");
-        } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/ByteUtil.java:491]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/HttpRequestLine.java [hunk 12]
```diff
@@ -67,11 +68,7 @@ public String getURI() {
     public String getBaseURI() {
         int idx = uri.indexOf('?');
         String baseURI = idx >= 0 ? uri.substring(0, idx) : uri;
-        try {
-            return URLDecoder.decode(baseURI, "utf-8");
-        } catch (UnsupportedEncodingException uee) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/HttpRequestLine.java:72]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/HttpUtil.java [hunk 13]
```diff
@@ -444,11 +445,8 @@ public static Map<String, String> getURIParams(String queryString) {
             String[] keyVal = pair.split("=");
             // URI query string is always encoded with application/x-www-form-urlencoded,
             // so use URLDecoder.decode() which converts '+' to ' '
-            try {
-                String value = keyVal.length > 1 ? URLDecoder.decode(keyVal[1], "UTF-8") : "";
-                params.put(URLDecoder.decode(keyVal[0], "UTF-8"), value);
-            } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/HttpUtil.java:450]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/HttpUtil.java [hunk 14]
```diff
@@ -544,15 +542,11 @@ public static String urlEscape(String str) {
                 if (escaped != null)
                     buf.append(escaped);
                 else {
-                    try {
-                        byte[] raw = Character.valueOf(c).toString().getBytes("UTF-8");
-                        for (byte b : raw) {
-                            int unsignedB = b & 0xFF;  // byte is signed
-                            buf.append("%").append(Integer.toHexString(unsignedB).toUpperCase());
-                        }
-                    } catch (IOException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/HttpUtil.java:553]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/HttpUtil.java [hunk 15]
```diff
@@ -600,11 +594,8 @@ public static String urlUnescape(String escaped) {
                 // append to the buffer if we are at the end of the string,
                 // or if this is the last encoded character in the segment.
                 if (i + 1 == escaped.length() || escaped.charAt(i + 1) != '%') {
-                    try {
-                        buf.append(new String(Bytes.toArray(segment), "UTF-8"));
-                    } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/HttpUtil.java:605]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/StringUtil.java [hunk 16]
```diff
@@ -979,20 +980,16 @@ public static String nonNull(String s) {
...
+        byte[] bodyBytes = body.getBytes(StandardCharsets.UTF_8);
+        if (maxBodyBytes > -1 && bodyBytes.length > maxBodyBytes) {
+            // During truncation make sure that we don't end up with a partial char at the end of the body.
+            // Start from index maxBodyBytes and going backwards determine the first byte that is a starting
+            // byte for a character. Such a byte is one whose top bit is 0 or whose top 2 bits are 11.
+            int indexToExclude = maxBodyBytes;
+            while (indexToExclude > 0 && bodyBytes[indexToExclude] < -64) {
+                indexToExclude--;
             }
-        } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/StringUtil.java:994]: @keshavbhatt check remove

File: common/src/main/java/com/zimbra/common/util/memcached/ZimbraMemcachedClient.java [hunk 17]
```diff
@@ -785,12 +785,7 @@ public boolean putBigByteArray(String key, byte[] value, int expirySec, long tim
             // Put the table of contents as the main value.  Do this after all chunks have been
             // added successfully.
             byte[] tocBytes;
-            try {
-                tocBytes = toc.encode().getBytes("utf-8");
-            } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline common/src/main/java/com/zimbra/common/util/memcached/ZimbraMemcachedClient.java:791]: @keshavbhatt check remove

File: store/src/main/java/com/zimbra/cs/account/DataSource.java [hunk 18]
```diff
@@ -447,11 +448,7 @@ private static Cipher getCipher(String dataSourceId, byte[] salt, boolean encryp
   }
 
   public static String encryptData(String dataSourceId, String data) throws ServiceException {
-    try {
-      return new String(encryptData(dataSourceId, data.getBytes("utf-8")));
-    } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline store/src/main/java/com/zimbra/cs/account/DataSource.java:452]: @keshavbhatt check remove

File: store/src/main/java/com/zimbra/cs/account/DataSource.java [hunk 19]
```diff
@@ -472,11 +469,7 @@ public static byte[] encryptData(String dataSourceId, byte[] data) throws Servic
   }
 
   public static String decryptData(String dataSourceId, String data) throws ServiceException {
-    try {
-      return new String(decryptData(dataSourceId, data.getBytes()), "utf-8");
-    } catch (UnsupportedEncodingException e) {
```
Comment (sonersivri) [inline store/src/main/java/com/zimbra/cs/account/DataSource.java:477]: @keshavbhatt check remove

File: client/src/main/java/com/zimbra/client/ZContactHit.java (hunk 1 above)
Comment (keshavbhatt) [reply client/src/main/java/com/zimbra/client/ZContactHit.java:304]: Fixed indentation.

File: client/src/main/java/com/zimbra/client/ZFolder.java (hunk 2 above)
Comment (keshavbhatt) [reply client/src/main/java/com/zimbra/client/ZFolder.java:593]: @sonersivri we checked, and we are fine with the new change.

File: client/src/main/java/com/zimbra/client/ZFolder.java (hunk 3 above)
Comment (keshavbhatt) [reply client/src/main/java/com/zimbra/client/ZFolder.java:644]: @sonersivri we found that the new change causes no issue.

File: common/src/main/java/com/zimbra/common/calendar/ZCalendar.java (hunk 4 above)
Comment (keshavbhatt) [reply common/src/main/java/com/zimbra/common/calendar/ZCalendar.java:1163]: Resolved IRL conversation.

File: common/src/main/java/com/zimbra/common/iochannel/Packet.java (hunk 5 above)
Comment (keshavbhatt) [reply common/src/main/java/com/zimbra/common/iochannel/Packet.java:62]: we do not need it anymore.

File: common/src/main/java/com/zimbra/common/iochannel/Packet.java (hunk 6 above)
Comment (keshavbhatt) [reply common/src/main/java/com/zimbra/common/iochannel/Packet.java:113]: We don't need it anymore.

File: common/src/main/java/com/zimbra/common/util/ByteUtil.java (hunk 10 above)
Comment (keshavbhatt) [reply common/src/main/java/com/zimbra/common/util/ByteUtil.java:476]: we dont need it anymore

File: store/src/main/java/com/zimbra/cs/account/DataSource.java (hunk 18 above)
Comment (keshavbhatt) [reply store/src/main/java/com/zimbra/cs/account/DataSource.java:452]: we don't need it anymore.

**Review by aheeva-yuliya — APPROVED** [review]

**Review by matteobaglini — APPROVED** [review]

**Pull request merged**